### PR TITLE
[Merged by Bors] - TY-2808 Implement age limit when fetching articles

### PR DESCRIPTION
**Pull request by x3ro**
[TY-2808]
Succesor of [TY-2784]

Mostly the same as https://github.com/xaynetwork/xayn_discovery_engine/tree/TY-2784-3-days-limit 

[TY-2808]: https://xainag.atlassian.net/browse/TY-2808?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TY-2784]: https://xainag.atlassian.net/browse/TY-2784?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by janpetschexain — APPROVED** [review]

File: discovery_engine_core/providers/src/client.rs [hunk 1]
```diff
@@ -212,6 +227,15 @@ impl Client {
...
 }
 
+/// Default `from` value for newscatcher news queries
+pub fn default_from() -> String {
+    let from = Utc::today() - chrono::Duration::days(3);
+    from.format("%Y/%m/%d").to_string()
+}
+
+/// Default `when` value for newscatcher headline queries
+pub const DEFAULT_WHEN: Option<&'static str> = Some("3d");
```
Comment (Robert-Steiner) [inline discovery_engine_core/providers/src/client.rs:237]: 🎨 we could make the type consistence with the return type of `default_from` by removing `Option`

```suggestion
pub const DEFAULT_WHEN: &str = "3d";
```

**Review by Robert-Steiner — APPROVED** [review]

File: discovery_engine_core/providers/src/client.rs [hunk 2]
```diff
@@ -103,6 +104,10 @@ pub struct NewsQuery<'a, F> {
     pub common: CommonQueryParts<'a>,
     /// News filter.
     pub filter: F,
+    /// Starting point in time from which to start the search.
+    /// The format is YYYY/mm/dd. Default timezone is UTC.
+    /// Defaults to the last week.
+    pub from: Option<String>,
```
[the remark this answers is not in the document]
Comment (x3ro) [reply discovery_engine_core/providers/src/client.rs:110]: I opted against this, as it makes the default value (the only one we have at the moment) more difficult to use, as we need to introduce an intermediate variable for ownership before the query is executed.

**Comment by x3ro** [comment]
bors merge

**Comment by x3ro** [comment]
bors merge-

**Comment by x3ro** [comment]
bors merge

**Pull request closed without merging**